src/components/PeopleToDisplay.tsx

```diff
@@ -0,0 +1,76 @@
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+
+export const PeopleToDisplay = (people: Person[]): Person[] => {
+  const [searchParams] = useSearchParams();
+
+  return useMemo(() => {
+    const queryFilter = searchParams.get('query')
+      ?.toLowerCase();
+    const sexFilter = searchParams.get('sex') || '';
+    const centuriesFilter = searchParams.getAll('centuries') || [];
+    const sortOrder = searchParams.get('sortOrder');
+    const sortBy = searchParams.get('sort');
```

Comment:
Filter/sort functions should be just helpers and shouldn't know anything about query params.